### PR TITLE
jewels and stones exercise

### PR DESCRIPTION
**Pull request by kcarrel**
Context from Leetcode: You're given strings J representing the types of stones that are jewels, and S representing the stones you have.  Each character in S is a type of stone you have.  You want to know how many of the stones you have are also jewels.

The letters in J are guaranteed distinct, and all characters in J and S are letters. Letters are case sensitive, so "a" is considered a different type of stone from "A".

### CODE REVIEW

File: src/golang.org/exercises/jewels-and-stones/jewels-and-stones.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"fmt"
+)
+
+func numJewelsInStones(J string, S string) int {
```
Comment (wadells) [inline src/golang.org/exercises/jewels-and-stones/jewels-and-stones.go]: When identifiers begin with a capital letter in Go, that typically means they are public and available for use outside the package.  Function parameters are always local and cannot be exported.  Thus I'd suggest making them lower case.

File: src/golang.org/exercises/jewels-and-stones/jewels-and-stones.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	jewels := make(map[string]struct{})
+	for i := range J {
+		jewels[string(J[i])] = struct{}{}
+	}
+
+	count := 0
+	for i := range S {
```
Comment (wadells) [inline src/golang.org/exercises/jewels-and-stones/jewels-and-stones.go]: Nit: Using `i` here is a touch misleading, as `i` typically refers to indices.  However this range iterates over the values and not the indices.

File: src/golang.org/exercises/jewels-and-stones/jewels-and-stones.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		if _, ok := jewels[string(S[i])]; ok {
+			count++
+		}
+	}
+	fmt.Printf("Count: %d \n", count)
```
Comment (wadells) [inline src/golang.org/exercises/jewels-and-stones/jewels-and-stones.go]: I'd leave IO (printing) to the caller.  For cleanliness, computational or library functions typically should not call print statements directly.  Either let the caller print or use a log statement.

File: src/golang.org/exercises/jewels-and-stones/jewels-and-stones.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+func main() {
+	var J = "aA"
+	var S = "aAAbbbb"
+	numJewelsInStones(J, S)
+	fmt.Printf("Expected count: 3 \n")
+
+	var J1 = "z"
+	var S1 = "ZZ"
+	numJewelsInStones(J1, S1)
+	fmt.Printf("Expected Count: 0 \n")
```
Comment (wadells) [inline src/golang.org/exercises/jewels-and-stones/jewels-and-stones.go:33]: Nice! I like that you included both a positive and a negative test case!

**Review by wadells — APPROVED** [review]

File: src/golang.org/exercises/jewels-and-stones/jewels-and-stones.go (hunk 1 above)
Comment (kcarrel) [reply src/golang.org/exercises/jewels-and-stones/jewels-and-stones.go]: Updated to lowercase in commit https://github.com/kcarrel/Golang-Training-Notes/pull/1/commits/0722bd650a57c4f9ee8e1a4801c15decc3142b5f

File: src/golang.org/exercises/jewels-and-stones/jewels-and-stones.go (hunk 3 above)
Comment (kcarrel) [reply src/golang.org/exercises/jewels-and-stones/jewels-and-stones.go]: moved printing outside of computational function in commit https://github.com/kcarrel/Golang-Training-Notes/pull/1/commits/0722bd650a57c4f9ee8e1a4801c15decc3142b5f

**Pull request merged**